### PR TITLE
School Library: Refactor code (Pair Programming)

### PR DESCRIPTION
**Pull request by mahdinoori2000**
### What is added

- YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- Followed SOLID principles.
- Pay to the "Single-responsibility Principle".
- The role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).
- Extract some pieces of logic into separate class(es).

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -65,6 +61,18 @@ def createperson
...
+    student = Student.new(age, name, false, nil) # Provide all four arguments
+    @peopl.push(student)
+    puts 'Person created successfully'
+  end
+
+  def having_parent_permission(age, name)
+    student = Student.new(age, name, true, nil)
+    @people.push(student)
+    puts 'Person created successfully'
+  end
```
Comment (skyv26) [inline app.rb]: - [ ] Please 🙏🏻 follow the ruby best practices, as you can see your functionality inside the two functions is similar except the parameters `true` and `false`. So please use the one function and re-use it again and again, which will help you to achieve more drier code.

File: app.rb
Comment (skyv26) [inline app.rb:1]: - [ ] Kindly follow the `YAGNI` principle according to which you should remove those things which are not useful and will not contribute to your project advancement. For line number **46** you are returning `nil` and the fact is your application is going to work without using this `keyword`.

**Review by skyv26 — CHANGES_REQUESTED** [review]
## Hi @mahdinoori2000 and @Samuelwanza , 

Good job so far 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## STATUS: REQUIRE CHANGES 🔴 



## Highlights
- Implement the features ✅ 
- Professional `README` template ✅ 
- Used correct `Gitflow` ✅ 
- Meaningful **Github** commits ✅ 
- Followed *best-practices* ✅ 
- Correct **Semantics** used ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AbelG101 — APPROVED** [review]
# Hi @mahdinoori2000  👋, 

**Your project is complete!  There is nothing else to say other than... it's time to merge it. 
Congratulations! 🎉 🥳🥇**

## Highlights
- DRY principles have been implemented  ✔
- Code follows best practices ✔
- Correct use of Git Flow ✔
- Linter check is passing ✔
- Descriptive PR title with a good summary ✔

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**